### DISCUSSION TITLE
a library alternative to document.execcommand

### DISCUSSION THREAD

**Discussion by aliibrahim123** [GrapesJS/grapesjs#5471] [General]
i am making a library called [simply.js](https://github.com/aliibrahim123/simply.js) which is a huge list of small independed modules featuring common utilities ranging from dom to data types to cli and encoding.   
and one of these modules is dom/edit.js which is a small document.executecommand alternative featuring most of its features.   
it is size is just 7.8 kb minified which is so low.   
i hope you think in integration it to the editor or to reverse enginearing so that the editor doesnt rely on an acient obselute method.

also there is a library integration it called [HyperRTE](https://github.com/aliibrahim123/hyperRTE) you can use it as a reference.
thanks for listening

**Comment by mosh-tudor** [comment]
> acient obselute method

Rly? :D

**Reply by aliibrahim123** [reply]
according to [mdn](https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand)

>  Deprecated: This feature is no longer recommended. Though some browsers might still support it, it may have already been removed from the relevant web standards, may be in the process of being dropped, or may only be kept for compatibility purposes. Avoid using it, and update existing code if possible; see the compatibility table at the bottom of this page to guide your decision. Be aware that this feature may cease to work at any time.

**Comment by artf** [comment]
Hi @aliibrahim123 you can try to [create the RTE plugin](https://grapesjs.com/docs/guides/Replace-Rich-Text-Editor.html) for your library
